pkg/tsdb/prometheus/querydata/request.go

```diff
@@ -112,19 +113,27 @@ func (s *QueryData) fetch(ctx context.Context, client *client.Client, q *models.
 		Error:  nil,
 	}
 
-	if q.RangeQuery {
-		res, err := s.rangeQuery(traceCtx, client, q, headers)
+	if q.InstantQuery {
+		res, err := s.instantQuery(traceCtx, client, q, headers)
 		if err != nil {
 			return nil, err
 		}
+		response.Error = res.Error
 		response.Frames = res.Frames
 	}
 
-	if q.InstantQuery {
-		res, err := s.instantQuery(traceCtx, client, q, headers)
+	if q.RangeQuery {
+		res, err := s.rangeQuery(traceCtx, client, q, headers)
 		if err != nil {
 			return nil, err
 		}
+		if res.Error != nil {
```

Comment:
it might be worth adding coverage in `testdata` for the `Both` error scenario

Reply:
It would be, but going to be take risk over time for two reasons:
 - Want to make sure it makes it to the next point release so we can start trying it out in cloud
 - Looking for this to be temporary, in that explore changes to sending the instant and range queries separately (and it gets removed from dashboard)  - https://github.com/grafana/grafana/issues/57365